src/movement/maestro.hpp

```diff
@@ -0,0 +1,342 @@
+             *        port to be used for communicating with the Maestro.
+             *
+             * @return Zero upon success and -1 upon failure.
+             */
+            int setPort(Serial *_port)
+            {
+                port = _port;
+                if (_port != NULL)
+                {
+                    uint8_t detect_byte = 0xAA;
+                    if (port->Write(&detect_byte, 1) != 1)
+                    {
+                        ROS_DEBUG_STREAM("Serial port failed to write baud-detection bit.");
+                        return -1;
+                    }
+                    return 0;
+                }
+                ROS_DEBUG_STREAM("Serial port pointer supplied is not a valid
+                        pointer.");
+                return -1;
+            }
+
+            /**
+             * Sets the speed of a particular thruster.
+             *
+             * @param speeds The normalized speeds to set all thrusters to.
+             *
+             * @return 0 on success and -1 on failure.
+             */
+            int set(vector<float> &speeds)
+            {
+                if (speeds.size() != thrusters || port == NULL) return -1;
+
+                uint8_t command[thrusters*2+3];
+                command[0] = static_cast<uint8_t>(MaestroCommands::SetMultipleTargets);
+                command[1] = thrusters;
+                command[2] = 0;
+
+                /*
+                 * Check to see if it's currently time to send a reset signal.
+                 */
+                ros::Time now = ros::Time::now();
+                if (now > nextReset)
+                {
+                    ROS_INFO_STREAM("Sending thruster reset signal.");
+                    for (int thruster = 0; thruster < thrusters; ++thruster)
+                    {
+                        if (parseNormalized(0, command[thruster*2+4],
+                                command[thruster*2+3]))
+                        {
+                            ROS_DEBUG_STREAM("Parse Normalized encountered
+                                    abnormal thruster speed.");
+                            return -1;
+                        }
+                    }
+
+                    if (port->Write(command, sizeof(command)) !=
+                            sizeof(command))
+                    {
+                        ROS_DEBUG_STREAM("Serial port failed to write entire
+                                command.");
+                        return -1;
+                    }
+
+                    nextReset = ros::Time::now() + ros::Duration(reset_delay);
+
+                    /*
+                     * Sleep for 185ms (9 50Hz cycles + 1/4 of one cycle) to
+                     * ensure that the zero pulse has propogated to the ESC.
+                     * Experimental tests have found this value. Any value less
+                     * is insufficient and will cause ESC malfuctions.
+                     */
+                    usleep(185000);
```

Comment:
Will add as parameter, since this is something that could potentially be updated later